src/Indicator.vala

```diff
@@ -161,6 +169,18 @@ public class Session.Indicator : Wingpanel.Indicator {
     }
 
     public override void closed () {}
+
+    private void show_shutdown_dialog (uint32 timestamp = 0) {
+        if (shutdown_dialog != null) {
+            shutdown_dialog.present ();
+            return;
+        }
+
```

Comment:
To simplify the code a little bit more, maybe don't make `shutdown_dialog` nullable, and initialize it directly in `construct` (and replace in destroy signal `shutdown_dialog = null;` with `shutdown_dialog.hide ()`?